pkg/registry/generic/etcd/etcd_test.go

```diff
@@ -343,254 +265,106 @@ func TestEtcdUpdate(t *testing.T) {
+	_, err := registry.Get(testContext, podA.Name)
+	if !errors.IsNotFound(err) {
+		t.Errorf("Unexpected error: %v", err)
 	}
 
-	key := "foo"
-
-	table := map[string]struct {
-		existing tools.EtcdResponseWithError
-		expect   runtime.Object
-		errOK    func(error) bool
-	}{
-		"normal": {
-			existing: nodeWithPodA,
-			expect:   podA,
-			errOK:    func(err error) bool { return err == nil },
-		},
-		"notExisting": {
-			existing: emptyNode,
-			expect:   nil,
-			errOK:    errors.IsNotFound,
-		},
+	registry.UpdateStrategy.(*testRESTStrategy).allowCreateOnUpdate = true
+	if !updateAndVerify(t, testContext, registry, podA) {
+		t.Errorf("Unexpected error updating podA")
 	}
 
-	testContext := api.WithNamespace(api.NewContext(), "test")
-
-	for name, item := range table {
-		fakeClient, registry := NewTestGenericEtcdRegistry(t)
-		path := etcdtest.AddPrefix("pods/foo")
-		fakeClient.Data[path] = item.existing
-		got, err := registry.Get(testContext, key)
-		if !item.errOK(err) {
-			t.Errorf("%v: unexpected error: %v", name, err)
-		}
-
-		if e, a := item.expect, got; !api.Semantic.DeepDerivative(e, a) {
-			t.Errorf("%v:\n%s", name, util.ObjectDiff(e, a))
-		}
-	}
 }
 
 func TestEtcdDelete(t *testing.T) {
 	podA := &api.Pod{
-		ObjectMeta: api.ObjectMeta{Name: "foo", ResourceVersion: "1"},
+		ObjectMeta: api.ObjectMeta{Name: "foo"},
 		Spec:       api.PodSpec{NodeName: "machine"},
 	}
 
-	nodeWithPodA := tools.EtcdResponseWithError{
-		R: &etcd.Response{
-			Node: &etcd.Node{
-				Value:         runtime.EncodeOrDie(testapi.Default.Codec(), podA),
-				ModifiedIndex: 1,
-				CreatedIndex:  1,
-			},
-		},
-		E: nil,
-	}
-
-	emptyNode := tools.EtcdResponseWithError{
-		R: &etcd.Response{},
-		E: tools.EtcdErrorNotFound,
-	}
-
 	testContext := api.WithNamespace(api.NewContext(), "test")
+	server, registry := NewTestGenericEtcdRegistry(t)
+	defer server.Terminate(t)
 
-	key := "foo"
+	// test failure condition
+	_, err := registry.Delete(testContext, podA.Name, nil)
+	if !errors.IsNotFound(err) {
+		t.Errorf("Unexpected error: %v", err)
+	}
 
-	table := map[string]struct {
-		existing tools.EtcdResponseWithError
-		expect   tools.EtcdResponseWithError
-		errOK    func(error) bool
-	}{
-		"normal": {
-			existing: nodeWithPodA,
-			expect:   emptyNode,
-			errOK:    func(err error) bool { return err == nil },
-		},
-		"notExisting": {
-			existing: emptyNode,
-			expect:   emptyNode,
-			errOK:    func(err error) bool { return errors.IsNotFound(err) },
-		},
+	// create pod
+	_, err = registry.Create(testContext, podA)
+	if err != nil {
+		t.Errorf("Unexpected error: %v", err)
 	}
 
-	for name, item := range table {
-		fakeClient, registry := NewTestGenericEtcdRegistry(t)
-		path := etcdtest.AddPrefix("pods/foo")
-		fakeClient.Data[path] = item.existing
-		obj, err := registry.Delete(testContext, key, nil)
-		if !item.errOK(err) {
-			t.Errorf("%v: unexpected error: %v (%#v)", name, err, obj)
-		}
+	// delete and check return object
```

Comment:
This seems like a bug - doesn't it?
If not - can you please update a comment to not state "and check return object"?